src/Bicep.LangServer/Handlers/TelemetryAndErrorHandlingHelper.cs

```diff
@@ -8,31 +8,35 @@
 
 namespace Bicep.LanguageServer.Handlers
 {
-    public class TelemetryAndErrorHandlingException : Exception
-    {
-        public BicepTelemetryEvent TelemetryEvent { get; }
-
-        public TelemetryAndErrorHandlingException(string message, BicepTelemetryEvent telemetryEvent)
-            : base(message)
-        {
-            TelemetryEvent = telemetryEvent;
-        }
-    }
-
     public class TelemetryAndErrorHandlingHelper<T>
     {
         private readonly IWindowLanguageServer window;
         private readonly ITelemetryProvider telemetryProvider;
-        private readonly T errorResponse;
+
```

Comment:
@anthony-c-martin I moved errorResponse to the exception rather than the helper constructor.

Reply:
See corresponding changes in Import Kubernetes and Insert Resource commands

Reply:
Although TBH haven't decided whether this pattern is very helpful.

Reply:
I created a new PR with the refactored decompile.  The telemetry changes are still there, if you're interested in reviewing them: https://github.com/Azure/bicep/pull/8804.  Thx.